pkg/util/encryption/encrypt.go

```diff
@@ -0,0 +1,99 @@
+package encryption
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"bytes"
+	"context"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+
+	"golang.org/x/crypto/chacha20poly1305"
+
+	"github.com/Azure/ARO-RP/pkg/env"
+)
+
+var (
+	Prefix        = []byte("ENC*")
+	_      Cipher = (*aeadCipher)(nil)
+)
+
+type Cipher interface {
+	Decrypt([]byte) ([]byte, error)
+	Encrypt([]byte) ([]byte, error)
+}
+
+type aeadCipher struct {
+	aead cipher.AEAD
+}
+
+func NewCipher(ctx context.Context, env env.Interface) (Cipher, error) {
+	keybase64, err := env.GetEncryptionSecret(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if keybase64 == nil {
+		return nil, fmt.Errorf("key not found")
+	}
+
+	key := make([]byte, 32)
```

Comment:
use DecodedLen here - avoids using the wrong secret, truncating it to 32 bytes and thinking it's OK to use.

Reply:
DecodeLen approximation is wrong. for 32 encoded it returns 33 decoded... 
Encoded is 44 bytes
```func (enc *Encoding) DecodedLen(n int) int {
	if enc.padChar == NoPadding {
		// Unpadded data may end with partial block of 2-3 characters.
		return n * 6 / 8  
	}
	// Padded base64 should always be a multiple of 4 characters in length.
	return n / 4 * 3
}
```

From what I understand it just approximation to bigger side, `44 * 6 / 8 = 33.... `